google/resource_dataflow_job.go

```diff
@@ -165,28 +166,40 @@ func resourceDataflowJobDelete(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return err
 	}
-	for _, ok := dataflowTerminalStatesMap[d.Get("state").(string)]; ok; _, ok = dataflowTerminalStatesMap[d.Get("state").(string)] {
+	for _, ok := dataflowTerminalStatesMap[d.Get("state").(string)]; !ok; _, ok = dataflowTerminalStatesMap[d.Get("state").(string)] {
```

Comment:
consider extracting `d.Get("state").(string)` to a local variable

Reply:
Done.

Reply:
Looks like you did so below, but not up here

Reply:
Ok I see it'll be different on the first read and on each successive pass, carry on